bsub -q voight_long -o chr20_ASW_AMR.out -e chr20_ASW_AMR.err 'vcftools --gzvcf ALL.chr20.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/ASW_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr20_ASW_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr20_MXL_AMR.out -e chr20_MXL_AMR.err 'vcftools --gzvcf ALL.chr20.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/MXL_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr20_MXL_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr20_PUR_AMR.out -e chr20_PUR_AMR.err 'vcftools --gzvcf ALL.chr20.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/PUR_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr20_PUR_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr20_CLM_AMR.out -e chr20_CLM_AMR.err 'vcftools --gzvcf ALL.chr20.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/CLM_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr20_CLM_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr20_PEL_AMR.out -e chr20_PEL_AMR.err 'vcftools --gzvcf ALL.chr20.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/PEL_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr20_PEL_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr19_ASW_AMR.out -e chr19_ASW_AMR.err 'vcftools --gzvcf ALL.chr19.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/ASW_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr19_ASW_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr19_ACB_AMR.out -e chr19_ACB_AMR.err 'vcftools --gzvcf ALL.chr19.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/ACB_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr19_ACB_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr19_MXL_AMR.out -e chr19_MXL_AMR.err 'vcftools --gzvcf ALL.chr19.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/MXL_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr19_MXL_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr19_PUR_AMR.out -e chr19_PUR_AMR.err 'vcftools --gzvcf ALL.chr19.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/PUR_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr19_PUR_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr19_CLM_AMR.out -e chr19_CLM_AMR.err 'vcftools --gzvcf ALL.chr19.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/CLM_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr19_CLM_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr19_PEL_AMR.out -e chr19_PEL_AMR.err 'vcftools --gzvcf ALL.chr19.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/PEL_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr19_PEL_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr18_ASW_AMR.out -e chr18_ASW_AMR.err 'vcftools --gzvcf ALL.chr18.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/ASW_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr18_ASW_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr18_ACB_AMR.out -e chr18_ACB_AMR.err 'vcftools --gzvcf ALL.chr18.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/ACB_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr18_ACB_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr18_MXL_AMR.out -e chr18_MXL_AMR.err 'vcftools --gzvcf ALL.chr18.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/MXL_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr18_MXL_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr18_PUR_AMR.out -e chr18_PUR_AMR.err 'vcftools --gzvcf ALL.chr18.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/PUR_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr18_PUR_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr18_CLM_AMR.out -e chr18_CLM_AMR.err 'vcftools --gzvcf ALL.chr18.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/CLM_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr18_CLM_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr18_PEL_AMR.out -e chr18_PEL_AMR.err 'vcftools --gzvcf ALL.chr18.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/PEL_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr18_PEL_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr17_ASW_AMR.out -e chr17_ASW_AMR.err 'vcftools --gzvcf ALL.chr17.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/ASW_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr17_ASW_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr17_ACB_AMR.out -e chr17_ACB_AMR.err 'vcftools --gzvcf ALL.chr17.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/ACB_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr17_ACB_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr17_MXL_AMR.out -e chr17_MXL_AMR.err 'vcftools --gzvcf ALL.chr17.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/MXL_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr17_MXL_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr17_PUR_AMR.out -e chr17_PUR_AMR.err 'vcftools --gzvcf ALL.chr17.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/PUR_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr17_PUR_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr17_CLM_AMR.out -e chr17_CLM_AMR.err 'vcftools --gzvcf ALL.chr17.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/CLM_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr17_CLM_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr17_PEL_AMR.out -e chr17_PEL_AMR.err 'vcftools --gzvcf ALL.chr17.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/PEL_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr17_PEL_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr16_ASW_AMR.out -e chr16_ASW_AMR.err 'vcftools --gzvcf ALL.chr16.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/ASW_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr16_ASW_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr16_ACB_AMR.out -e chr16_ACB_AMR.err 'vcftools --gzvcf ALL.chr16.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/ACB_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr16_ACB_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr16_MXL_AMR.out -e chr16_MXL_AMR.err 'vcftools --gzvcf ALL.chr16.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/MXL_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr16_MXL_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr16_PUR_AMR.out -e chr16_PUR_AMR.err 'vcftools --gzvcf ALL.chr16.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/PUR_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr16_PUR_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr16_CLM_AMR.out -e chr16_CLM_AMR.err 'vcftools --gzvcf ALL.chr16.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/CLM_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr16_CLM_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr16_PEL_AMR.out -e chr16_PEL_AMR.err 'vcftools --gzvcf ALL.chr16.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/PEL_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr16_PEL_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr15_ASW_AMR.out -e chr15_ASW_AMR.err 'vcftools --gzvcf ALL.chr15.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/ASW_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr15_ASW_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr15_ACB_AMR.out -e chr15_ACB_AMR.err 'vcftools --gzvcf ALL.chr15.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/ACB_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr15_ACB_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr15_MXL_AMR.out -e chr15_MXL_AMR.err 'vcftools --gzvcf ALL.chr15.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/MXL_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr15_MXL_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr15_PUR_AMR.out -e chr15_PUR_AMR.err 'vcftools --gzvcf ALL.chr15.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/PUR_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr15_PUR_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr15_CLM_AMR.out -e chr15_CLM_AMR.err 'vcftools --gzvcf ALL.chr15.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/CLM_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr15_CLM_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr15_PEL_AMR.out -e chr15_PEL_AMR.err 'vcftools --gzvcf ALL.chr15.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/PEL_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr15_PEL_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr14_ASW_AMR.out -e chr14_ASW_AMR.err 'vcftools --gzvcf ALL.chr14.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/ASW_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr14_ASW_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr14_ACB_AMR.out -e chr14_ACB_AMR.err 'vcftools --gzvcf ALL.chr14.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/ACB_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr14_ACB_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr14_MXL_AMR.out -e chr14_MXL_AMR.err 'vcftools --gzvcf ALL.chr14.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/MXL_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr14_MXL_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr14_PUR_AMR.out -e chr14_PUR_AMR.err 'vcftools --gzvcf ALL.chr14.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/PUR_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr14_PUR_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr14_CLM_AMR.out -e chr14_CLM_AMR.err 'vcftools --gzvcf ALL.chr14.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/CLM_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr14_CLM_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr14_PEL_AMR.out -e chr14_PEL_AMR.err 'vcftools --gzvcf ALL.chr14.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/PEL_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr14_PEL_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr13_ASW_AMR.out -e chr13_ASW_AMR.err 'vcftools --gzvcf ALL.chr13.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/ASW_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr13_ASW_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr13_ACB_AMR.out -e chr13_ACB_AMR.err 'vcftools --gzvcf ALL.chr13.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/ACB_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr13_ACB_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr13_MXL_AMR.out -e chr13_MXL_AMR.err 'vcftools --gzvcf ALL.chr13.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/MXL_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr13_MXL_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr13_PUR_AMR.out -e chr13_PUR_AMR.err 'vcftools --gzvcf ALL.chr13.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/PUR_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr13_PUR_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr13_CLM_AMR.out -e chr13_CLM_AMR.err 'vcftools --gzvcf ALL.chr13.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/CLM_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr13_CLM_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr13_PEL_AMR.out -e chr13_PEL_AMR.err 'vcftools --gzvcf ALL.chr13.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/PEL_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr13_PEL_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr12_ASW_AMR.out -e chr12_ASW_AMR.err 'vcftools --gzvcf ALL.chr12.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/ASW_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr12_ASW_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr12_ACB_AMR.out -e chr12_ACB_AMR.err 'vcftools --gzvcf ALL.chr12.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/ACB_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr12_ACB_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr12_MXL_AMR.out -e chr12_MXL_AMR.err 'vcftools --gzvcf ALL.chr12.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/MXL_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr12_MXL_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr12_PUR_AMR.out -e chr12_PUR_AMR.err 'vcftools --gzvcf ALL.chr12.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/PUR_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr12_PUR_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr12_CLM_AMR.out -e chr12_CLM_AMR.err 'vcftools --gzvcf ALL.chr12.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/CLM_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr12_CLM_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr12_PEL_AMR.out -e chr12_PEL_AMR.err 'vcftools --gzvcf ALL.chr12.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/PEL_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr12_PEL_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr11_ASW_AMR.out -e chr11_ASW_AMR.err 'vcftools --gzvcf ALL.chr11.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/ASW_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr11_ASW_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr11_ACB_AMR.out -e chr11_ACB_AMR.err 'vcftools --gzvcf ALL.chr11.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/ACB_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr11_ACB_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr11_MXL_AMR.out -e chr11_MXL_AMR.err 'vcftools --gzvcf ALL.chr11.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/MXL_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr11_MXL_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr11_PUR_AMR.out -e chr11_PUR_AMR.err 'vcftools --gzvcf ALL.chr11.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/PUR_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr11_PUR_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr11_CLM_AMR.out -e chr11_CLM_AMR.err 'vcftools --gzvcf ALL.chr11.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/CLM_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr11_CLM_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr11_PEL_AMR.out -e chr11_PEL_AMR.err 'vcftools --gzvcf ALL.chr11.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/PEL_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr11_PEL_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr10_ASW_AMR.out -e chr10_ASW_AMR.err 'vcftools --gzvcf ALL.chr10.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/ASW_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr10_ASW_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr10_ACB_AMR.out -e chr10_ACB_AMR.err 'vcftools --gzvcf ALL.chr10.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/ACB_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr10_ACB_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr10_MXL_AMR.out -e chr10_MXL_AMR.err 'vcftools --gzvcf ALL.chr10.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/MXL_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr10_MXL_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr10_PUR_AMR.out -e chr10_PUR_AMR.err 'vcftools --gzvcf ALL.chr10.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/PUR_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr10_PUR_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr10_CLM_AMR.out -e chr10_CLM_AMR.err 'vcftools --gzvcf ALL.chr10.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/CLM_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr10_CLM_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr10_PEL_AMR.out -e chr10_PEL_AMR.err 'vcftools --gzvcf ALL.chr10.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/PEL_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr10_PEL_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr9_ASW_AMR.out -e chr9_ASW_AMR.err 'vcftools --gzvcf ALL.chr9.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/ASW_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr9_ASW_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr9_ACB_AMR.out -e chr9_ACB_AMR.err 'vcftools --gzvcf ALL.chr9.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/ACB_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr9_ACB_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr9_MXL_AMR.out -e chr9_MXL_AMR.err 'vcftools --gzvcf ALL.chr9.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/MXL_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr9_MXL_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr9_PUR_AMR.out -e chr9_PUR_AMR.err 'vcftools --gzvcf ALL.chr9.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/PUR_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr9_PUR_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr9_CLM_AMR.out -e chr9_CLM_AMR.err 'vcftools --gzvcf ALL.chr9.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/CLM_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr9_CLM_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr9_PEL_AMR.out -e chr9_PEL_AMR.err 'vcftools --gzvcf ALL.chr9.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/PEL_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr9_PEL_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr8_ASW_AMR.out -e chr8_ASW_AMR.err 'vcftools --gzvcf ALL.chr8.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/ASW_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr8_ASW_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr8_ACB_AMR.out -e chr8_ACB_AMR.err 'vcftools --gzvcf ALL.chr8.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/ACB_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr8_ACB_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr8_MXL_AMR.out -e chr8_MXL_AMR.err 'vcftools --gzvcf ALL.chr8.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/MXL_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr8_MXL_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr8_PUR_AMR.out -e chr8_PUR_AMR.err 'vcftools --gzvcf ALL.chr8.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/PUR_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr8_PUR_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr8_CLM_AMR.out -e chr8_CLM_AMR.err 'vcftools --gzvcf ALL.chr8.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/CLM_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr8_CLM_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr8_PEL_AMR.out -e chr8_PEL_AMR.err 'vcftools --gzvcf ALL.chr8.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/PEL_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr8_PEL_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr7_ASW_AMR.out -e chr7_ASW_AMR.err 'vcftools --gzvcf ALL.chr7.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/ASW_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr7_ASW_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr7_ACB_AMR.out -e chr7_ACB_AMR.err 'vcftools --gzvcf ALL.chr7.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/ACB_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr7_ACB_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr7_MXL_AMR.out -e chr7_MXL_AMR.err 'vcftools --gzvcf ALL.chr7.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/MXL_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr7_MXL_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr7_PUR_AMR.out -e chr7_PUR_AMR.err 'vcftools --gzvcf ALL.chr7.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/PUR_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr7_PUR_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr7_CLM_AMR.out -e chr7_CLM_AMR.err 'vcftools --gzvcf ALL.chr7.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/CLM_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr7_CLM_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr7_PEL_AMR.out -e chr7_PEL_AMR.err 'vcftools --gzvcf ALL.chr7.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/PEL_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr7_PEL_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr6_ASW_AMR.out -e chr6_ASW_AMR.err 'vcftools --gzvcf ALL.chr6.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/ASW_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr6_ASW_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr6_ACB_AMR.out -e chr6_ACB_AMR.err 'vcftools --gzvcf ALL.chr6.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/ACB_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr6_ACB_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr6_MXL_AMR.out -e chr6_MXL_AMR.err 'vcftools --gzvcf ALL.chr6.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/MXL_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr6_MXL_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr6_PUR_AMR.out -e chr6_PUR_AMR.err 'vcftools --gzvcf ALL.chr6.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/PUR_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr6_PUR_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr6_CLM_AMR.out -e chr6_CLM_AMR.err 'vcftools --gzvcf ALL.chr6.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/CLM_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr6_CLM_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr6_PEL_AMR.out -e chr6_PEL_AMR.err 'vcftools --gzvcf ALL.chr6.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/PEL_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr6_PEL_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr5_ASW_AMR.out -e chr5_ASW_AMR.err 'vcftools --gzvcf ALL.chr5.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/ASW_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr5_ASW_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr5_ACB_AMR.out -e chr5_ACB_AMR.err 'vcftools --gzvcf ALL.chr5.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/ACB_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr5_ACB_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr5_MXL_AMR.out -e chr5_MXL_AMR.err 'vcftools --gzvcf ALL.chr5.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/MXL_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr5_MXL_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr5_PUR_AMR.out -e chr5_PUR_AMR.err 'vcftools --gzvcf ALL.chr5.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/PUR_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr5_PUR_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr5_CLM_AMR.out -e chr5_CLM_AMR.err 'vcftools --gzvcf ALL.chr5.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/CLM_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr5_CLM_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr5_PEL_AMR.out -e chr5_PEL_AMR.err 'vcftools --gzvcf ALL.chr5.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/PEL_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr5_PEL_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr4_ASW_AMR.out -e chr4_ASW_AMR.err 'vcftools --gzvcf ALL.chr4.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/ASW_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr4_ASW_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr4_ACB_AMR.out -e chr4_ACB_AMR.err 'vcftools --gzvcf ALL.chr4.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/ACB_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr4_ACB_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr4_MXL_AMR.out -e chr4_MXL_AMR.err 'vcftools --gzvcf ALL.chr4.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/MXL_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr4_MXL_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr4_PUR_AMR.out -e chr4_PUR_AMR.err 'vcftools --gzvcf ALL.chr4.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/PUR_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr4_PUR_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr4_CLM_AMR.out -e chr4_CLM_AMR.err 'vcftools --gzvcf ALL.chr4.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/CLM_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr4_CLM_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr4_PEL_AMR.out -e chr4_PEL_AMR.err 'vcftools --gzvcf ALL.chr4.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/PEL_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr4_PEL_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr3_ASW_AMR.out -e chr3_ASW_AMR.err 'vcftools --gzvcf ALL.chr3.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/ASW_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr3_ASW_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr3_ACB_AMR.out -e chr3_ACB_AMR.err 'vcftools --gzvcf ALL.chr3.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/ACB_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr3_ACB_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr3_MXL_AMR.out -e chr3_MXL_AMR.err 'vcftools --gzvcf ALL.chr3.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/MXL_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr3_MXL_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr3_PUR_AMR.out -e chr3_PUR_AMR.err 'vcftools --gzvcf ALL.chr3.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/PUR_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr3_PUR_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr3_CLM_AMR.out -e chr3_CLM_AMR.err 'vcftools --gzvcf ALL.chr3.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/CLM_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr3_CLM_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr3_PEL_AMR.out -e chr3_PEL_AMR.err 'vcftools --gzvcf ALL.chr3.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/PEL_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr3_PEL_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr2_ASW_AMR.out -e chr2_ASW_AMR.err 'vcftools --gzvcf ALL.chr2.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/ASW_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr2_ASW_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr2_ACB_AMR.out -e chr2_ACB_AMR.err 'vcftools --gzvcf ALL.chr2.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/ACB_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr2_ACB_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr2_MXL_AMR.out -e chr2_MXL_AMR.err 'vcftools --gzvcf ALL.chr2.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/MXL_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr2_MXL_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr2_PUR_AMR.out -e chr2_PUR_AMR.err 'vcftools --gzvcf ALL.chr2.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/PUR_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr2_PUR_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr2_CLM_AMR.out -e chr2_CLM_AMR.err 'vcftools --gzvcf ALL.chr2.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/CLM_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr2_CLM_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr2_PEL_AMR.out -e chr2_PEL_AMR.err 'vcftools --gzvcf ALL.chr2.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/PEL_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr2_PEL_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr1_ASW_AMR.out -e chr1_ASW_AMR.err 'vcftools --gzvcf ALL.chr1.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/ASW_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr1_ASW_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr1_ACB_AMR.out -e chr1_ACB_AMR.err 'vcftools --gzvcf ALL.chr1.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/ACB_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr1_ACB_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr1_MXL_AMR.out -e chr1_MXL_AMR.err 'vcftools --gzvcf ALL.chr1.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/MXL_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr1_MXL_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr1_PUR_AMR.out -e chr1_PUR_AMR.err 'vcftools --gzvcf ALL.chr1.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/PUR_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr1_PUR_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr1_CLM_AMR.out -e chr1_CLM_AMR.err 'vcftools --gzvcf ALL.chr1.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/CLM_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr1_CLM_AMR_multiallelic.vcf.gz'
bsub -q voight_long -o chr1_PEL_AMR.out -e chr1_PEL_AMR.err 'vcftools --gzvcf ALL.chr1.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep /project/voight_MR/raikens/SNP_AFs/populations/PEL_AMR --bed /project/voight_MR/raikens/SNP_AFs/nc_regions  --mac 2 --remove-indels --remove-filtered-all --recode --recode-INFO AA --stdout | bgzip -c > /project/voight_MR/raikens/SNP_AFs/chr1_PEL_AMR_multiallelic.vcf.gz'
