---
title: "Basic Plots and Data Gazing"
author: "Rachael 'Rocky' Aikens, Voight Lab"
date: "July 6, 2017"
output:
  pdf_document:
    latex_engine: xelatex
  html_document: default
---

```{r setup, echo = F, message = F, warning=F}
library(readr)
library(knitr)
library(ggplot2)
library(dplyr)
require(reshape2)
library(gridExtra)
library(grid)
require("ggrepel")
```


This document contains notes and visualizations of mutation rates for patterns of mutation on the 3mer, 5mer, and 7mer level which are interesting either because of their significance in heterogeneity test or the way that they cluster together in heatmaps.

```{r upload 7mers, echo = F, message = F}
setwd("C:/Users/VoightLab/Dropbox/SNP_rates/Raw_Data/singletons_excluded/")
AFR_7mer_counts <- read_delim("7mer/AFR_7mer_counts.txt", 
                              "\t", escape_double = FALSE, trim_ws = TRUE)
EUR_7mer_counts <- read_delim("7mer/EUR_7mer_counts.txt", 
                              "\t", escape_double = FALSE, trim_ws = TRUE)
EAS_7mer_counts <- read_delim("7mer/EAS_7mer_counts.txt", 
                              "\t", escape_double = FALSE, trim_ws = TRUE)
SAS_7mer_counts <- read_delim("7mer/SAS_7mer_counts.txt", 
                              "\t", escape_double = FALSE, trim_ws = TRUE)
```

```{r upload 5mers, echo = F, message = F}
setwd("C:/Users/VoightLab/Dropbox/SNP_rates/Raw_Data/singletons_excluded/")
AFR_5mer_counts <- read_delim("5mer/AFR_5mer_counts.txt", 
                              "\t", escape_double = FALSE, trim_ws = TRUE)
EUR_5mer_counts <- read_delim("5mer/EUR_5mer_counts.txt", 
                              "\t", escape_double = FALSE, trim_ws = TRUE)
EAS_5mer_counts <- read_delim("5mer/EAS_5mer_counts.txt", 
                              "\t", escape_double = FALSE, trim_ws = TRUE)
SAS_5mer_counts <- read_delim("5mer/SAS_5mer_counts.txt", 
                              "\t", escape_double = FALSE, trim_ws = TRUE)
```

```{r upload 3mers, echo = F, message = F}
setwd("C:/Users/VoightLab/Dropbox/SNP_rates/Raw_Data/singletons_excluded/")
AFR_3mer_counts <- read_delim("3mer/AFR_3mer_counts.txt", 
                              "\t", escape_double = FALSE, trim_ws = TRUE)
EUR_3mer_counts <- read_delim("3mer/EUR_3mer_counts.txt", 
                              "\t", escape_double = FALSE, trim_ws = TRUE)
EAS_3mer_counts <- read_delim("3mer/EAS_3mer_counts.txt", 
                              "\t", escape_double = FALSE, trim_ws = TRUE)
SAS_3mer_counts <- read_delim("3mer/SAS_3mer_counts.txt", 
                              "\t", escape_double = FALSE, trim_ws = TRUE)
```

```{r upload gw counts, echo = F, message = F}
setwd("C:/Users/VoightLab/Dropbox/SNP_rates/Raw_Data/gw_counts/")

gw_5mer_counts <- read_delim("gw_5mer_counts.txt", 
                             "\t", escape_double = FALSE, trim_ws = TRUE)
gw_3mer_counts <- read_delim("gw_3mer_counts.txt", 
                             "\t", escape_double = FALSE, trim_ws = TRUE)
gw_7mer_counts <- read_delim("gw_7mer_counts.txt", 
                             "\t", escape_double = FALSE, trim_ws = TRUE)
```

```{r upload subpops, echo = F, eval = F, message = F}
source("C:/Users/VoightLab/Dropbox/SNP_rates/Code/data_wrangling/upload_subpops_3mer.R")
source("C:/Users/VoightLab/Dropbox/SNP_rates/Code/data_wrangling/upload_subpops_5mer.R")
source("C:/Users/VoightLab/Dropbox/SNP_rates/Code/data_wrangling/upload_subpops_7mer.R")
```


# Methodology

I use a handful of different plotting methods to gain different perpectives on the data.  These are:

* **CI.plot** Given count dataframes for four populations and a polymorphism of interest $m$, plot the inferred mutation rates of $m$ in each population with approximate confidence intervals.

```{r def CI plot, echo = F}
CI.plot <- function(AFR, EUR, EAS, SAS, mut) {
  
  mut.i <- which(AFR$Context == mut)
  popnames <- c("Africa","Europe", "South\nAsia", "East\nAsia")
  colors <- c("forestgreen", "darkblue", "magenta","red")
  
  #have to do a silly workaround or R will sort popnames alphanumerically
  poplabs <- factor(popnames, levels= popnames)
  
  #cycle through pops and get counts for mut
  counts <- rep(0, 4)
  sums <- rep(0, 4)
  pops <- list(AFR, EUR, SAS, EAS)
  
  for (i in 1:length(pops)){
    counts[i] <- pops[[i]]$Count[mut.i]
    sums[i]<- sum(pops[[i]]$Count)
  }
  
  #estimate substitution probability
  N.c <- AFR$context_in_genome[mut.i] 
  theta <- counts/N.c
  L <- theta - 1.96*sqrt(theta*(1-theta)/N.c)
  U <- theta + 1.96*sqrt(theta*(1-theta)/N.c)
  
  #normalize to rate estimate; assume genome wide subsitution probability is measured without error
  norm <- 1.2E-8*(sum(as.numeric(AFR$context_in_genome))/3)/sums
  
  df <- data.frame(cbind(popnames, theta*norm, L*norm, U*norm))
  plotcol <- reorder(colors, theta*norm)
  
  CI.plot <- ggplot(df, aes(reorder(popnames, c(1,2,3,4)), theta*norm)) +
    geom_point(size = 4, color = plotcol) +
    geom_errorbar(aes(ymax = U*norm, ymin = L*norm), color = plotcol, size =.75)+
    labs(title = paste("Estimated mutation\nrate of", mut), y = NULL) + #y axis label
    theme(axis.text.x = element_text(size = rel(.9)), axis.title.x = element_blank(), #adjust text sizes
          axis.title.y = element_text(size = rel(.9)), axis.text.y = element_text(size = rel(.9), angle = 0, hjust = 0.5), title = element_text(size = rel(.7)),
          legend.position = 'none')
  return(CI.plot)
}
```

```{r, def CI.plot.subpop, echo = F}
colors.ancestral <- c(rep("forestgreen", 5), rep("darkblue", 5), 
                      rep("magenta", 5), rep("red", 5))
colors.all <- c(rep("forestgreen", 5), rep("orange", 6), 
                rep("darkblue", 5), rep("magenta", 5), rep("red", 5))


CI.plot.subpop <- function(subpops, mut, popnames, colors = colors.ancestral){
  n.pops <- length(subpops)
  mut.i <- which(subpops[[1]]$Context == mut)
  
  #have to do a silly workaround or R will sort popnames alphanumerically
  poplabs <- factor(popnames, levels= popnames)
  
  #cycle through subpops and get counts for mut
  counts <- rep(0, n.pops)
  sums <- rep(0, n.pops)
  for (i in 1:n.pops){
    counts[i] <- subpops[[i]]$Count[mut.i]
    sums[i]<- sum(subpops[[i]]$Count)
  }
  
  #estimate substitution probability
  N.c <- subpops[[1]]$context_in_genome[mut.i] 
  theta <- counts/N.c
  L <- theta - 1.96*sqrt(theta*(1-theta)/N.c)
  U <- theta + 1.96*sqrt(theta*(1-theta)/N.c)
  
  #normalize to rate estimate; assume genome wide subsitution probability is measured without error
  norm <- 1.2E-8*(sum(as.numeric(subpops[[1]]$context_in_genome))/3)/sums
  
  
  df <- data.frame(cbind(poplabs, theta*norm, L*norm, U*norm))
  plotcol <- reorder(colors, theta*norm)
  
  ggplot(df, aes(reorder(popnames, poplabs), theta*norm)) +
    geom_point(size = 4, color = plotcol) +
    geom_errorbar(aes(ymax = U*norm, ymin = L*norm), color = plotcol, size = 0.75)+
    labs(y = paste("Estimated mutation rate of ", mut,"\n")) + #y axis label
    theme(axis.text.x = element_text(size = rel(1), angle = 45, hjust = 1), axis.title.x = element_blank(), #adjust text sizes
          axis.title.y = element_text(size = rel(1)), axis.text.y = element_text(size = rel(.9)),
          legend.position = 'none')
}
```

* **chrom.box** Given count dataframes for four populations, a polymorphism of interest $m$, and a dataframe of genome wide context counts, plot the inferred mutation rates of $m$ in each on each chromosome as a boxplot, labeling outliers.

```{r def chrom.box, echo = F}
#box plot of rate by chrom for a particular sequence context
chrom.box <- function(AFR, EUR, EAS, SAS, gw, mut){
  #get data
  chrom.dat <- chrom.process.data(AFR, EUR, EAS, SAS, gw, mut)
  
  #find outliers; THIS DOESN'T WORK :(
  AFR.rates <- subset(chrom.dat, chrom.dat$pop == "Africa")
  AFR.outliers <- subset(AFR.rates, is_outlier(AFR.rates$rate)==TRUE)
  EUR.rates <- subset(chrom.dat, chrom.dat$pop == "Europe")
  EUR.outliers <- subset(EUR.rates, is_outlier(EUR.rates$rate)==TRUE)
  EAS.rates <- subset(chrom.dat, chrom.dat$pop == "East\nAsia")
  EAS.outliers <- subset(EAS.rates, is_outlier(EAS.rates$rate)==TRUE)
  SAS.rates <- subset(chrom.dat, chrom.dat$pop == "South\nAsia")
  SAS.outliers <- subset(SAS.rates, is_outlier(SAS.rates$rate)==TRUE)
  
  #plot
  c_plot <- ggplot(chrom.dat, aes(pop, rate))+
    geom_boxplot(outlier.color = NA, fill = c("palegreen1",  "steelblue1", 'plum1', "lightcoral"))+
    labs(x = "\nPopulation", title = paste("Mutation rate of", mut,"\nby chromosome"), y = NULL)+
    scale_color_manual("", values = c("forest green", "dark blue", 'magenta', "red"))+
    
    #add outlier labels
    geom_text(data = EUR.outliers, aes(pop, rate, label = chrom), color = "dark blue", nudge_x = 0.1)+
    geom_text(data = AFR.outliers, aes(pop, rate, label = chrom), color = "forest green", nudge_x = 0.1)+
    geom_text(data = EAS.outliers, aes(pop, rate, label = chrom), color = "red", nudge_x = 0.1)+
    geom_text(data = SAS.outliers, aes(pop, rate, label = chrom), color = "magenta", nudge_x = 0.1)+
    
    #add points
    geom_point(aes(color = factor(chrom.dat$pop)), size = 1.2, position = position_jitter(width = 0.1))+
    theme(axis.text.x = element_text(size = rel(.9)), axis.title.x = element_blank(), #adjust text sizes
          axis.title.y = element_text(size = rel(.9)), axis.text.y = element_text(size = rel(.9)),
          legend.position = 'none',
          title = element_text(size = rel(.7)))#legend position
  return(c_plot)
}

#helper function for ploting which formats the data for ggplot
chrom.process.data <- function(AFR, EUR, EAS, SAS, gw, mut){
  #get indicies for mutation and contextAFR, EUR, EAS, SAS, gw, mut
  i <- which(EUR$Context == mut)
  cntxt <- substr(mut, 1, nchar(mut)-3)
  i.cntxt <- which(gw$Context == cntxt)
  col.e <- ncol(AFR)
  col.s <- col.e-22
  
  #trim summary columns from each input dataframe
  AFR <- AFR[c(col.s: col.e)]
  EUR <- EUR[c(col.s: col.e)]
  EAS <- EAS[c(col.s: col.e)]
  SAS <- SAS[c(col.s: col.e)]
  gw <- gw[-c(1,2)]
  gw_totals <- colSums(gw)
  
  #make output dataframe for plot
  dat <- data.frame(matrix(nrow = 23, ncol = 4))
  colnames(dat) <- c("Africa", "Europe", "South\nAsia", "East\nAsia")  
  
  #get rates for each pop
  dat$'Europe' <- t(EUR[i,]/gw[i.cntxt,]*1.2E-8*gw_totals/colSums(EUR))
  dat$'Africa' <- t(AFR[i,]/gw[i.cntxt,]*1.2E-8*gw_totals/colSums(AFR))
  dat$'East\nAsia' <- t(EAS[i,]/gw[i.cntxt,]*1.2E-8*gw_totals/colSums(EAS))
  dat$'South\nAsia' <- t(SAS[i,]/gw[i.cntxt,]*1.2E-8*gw_totals/colSums(SAS))
  
  #melt data to necessary format
  row.names(dat) <- c('1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12', '13', '14', '15', '16', '17', '18', '19', '20', '21', '22', 'X')
  dat.m <- melt(t(dat))
  colnames(dat.m) <- c("pop", "chrom", "rate")
  
  return(dat.m)
}

#helper function that returns which elements of a vector x are outliers
is_outlier <- function(x) {
  return(x < quantile(x, 0.25) - 1.5 * IQR(x) | x > quantile(x, 0.75) + 1.5 * IQR(x))
}
```

* **subrate.scplot** Given count dataframes for two populations and a 3mer polymorphism type, find the rates of all expansions of that threemer in those populations and plot them against each other.

```{r pairwise.chi, echo = F}
# calculates homogeneity test p values for pairwise comparisons of two dfs of counts
pairwise.chi <- function(counts.1, counts.2, filter = T){
  n.contexts = length(counts.1$Context)
  result <- data.frame(matrix(ncol=2,nrow=n.contexts))
  colnames(result) <- c("Context", "p")
  
  result$Context <- counts.1$Context
  sum.1 <- sum(counts.1$Count)
  sum.2 <- sum(counts.2$Count)

  for (i in 1:n.contexts){
    c.a <- c(counts.1$Count[i], counts.2$Count[i])
    c.b <- c(sum.1, sum.2) - c.a
    data <- cbind(c.a, c.b)
    warning <- is(tryCatch(chisq.test(data), warning = function(w) w), "warning")
    if (filter == T & warning){
      result$p[i] <- NA
    }
    else result$p[i] <- chisq.test(data)$p.value
  }
  
  return(result)
}
```

```{r def subrate scplot, echo=F}
# given: count1 and count2: pop-specific polymorphism dataframes; Mut: 3mer subcontext of interest,
# generates a scatter plot of rate for mut in population 1 by population 2
subrate.scplot <- function(count1, count2, mut, pops = c("Pop 1", "Pop 2"), label = c(), lm = T){
  pop1.mut <- subset(count1, count1$X3mer == mut)
  pop2.mut <- subset(count2, count2$X3mer == mut)
  sample.size <- pop1.mut$Count + pop2.mut$Count
  
  #make plot
  p_plot <- qplot(pop1.mut$Rate, pop2.mut$Rate) +
    labs(x = paste(mut, " in ", pops[1]), y = paste(mut, " in ", pops[2])) +
    theme(legend.title = element_blank(), 
          legend.justification= c(1,0), 
          legend.position=c(1,0),
          legend.text = element_text(size = rel(0.75)),
          legend.key.size = unit(.4, "cm"),
          axis.text=element_text(size=8),
          text = element_text(size = 9))+
    geom_point(aes(colour = log(base = 10, x = sample.size)), size = 1.25) + 
    scale_colour_gradient(low = "orange", high = "blue")+
    geom_abline(aes(intercept=0,slope=1), size = .5) + 
    coord_fixed(ratio = 1, 
                xlim = c(0, max(pop1.mut$Rate, pop2.mut$Rate)),
                ylim = c(0, max(pop1.mut$Rate, pop2.mut$Rate)))
  
  if (lm){
    mod1 <- lm(pop2.mut$Rate ~ pop1.mut$Rate)
    pred <- predict(mod1, interval = "prediction")
    
    p_plot <- p_plot + stat_smooth(method = lm, se = FALSE, fullrange = T)+
      geom_line(aes(y = pred[,2]), color = "red", linetype = "dashed", size = 1)+
      geom_line(aes(y = pred[,3]), color = "red", linetype = "dashed", size = 1)
  }
  
  if (length(label!=0)){
    #set aside contexts to label
    dat <- data.frame(cbind(pop1.mut$Context, pop1.mut$Rate, pop2.mut$Rate), stringsAsFactors = F)
    lab <- subset(dat,  is.element(dat[,1], label))
    
    if (length(lab[,1]!=0)){
      lab[,c(2,3)] <- sapply(lab[,c(2,3)], as.numeric)
      p_plot <- p_plot + geom_label_repel(data = lab, aes(lab[,2], lab[,3],
                                   label = lab[,1]), size = 2.5, box.padding = unit(1, "lines"),
                                   point.padding = unit(.3, "lines"),
                                   nudge_x = 1.5e-9, nudge_y = 1.5e-9, color = "purple4")
    }
  }

  return(p_plot)
}
```


* **subrate.lplot** Given count dataframes for each non-admixed continental group and a 3mer polymorphism type, find the rates of all 5mer expansions of that threemer in those populations and plot them as as lines across the populations.

```{r def subrate lplot, echo=F}
# given: count files for each pop and a given 3mer type, 
# plots all 5mers with that 3mer subcontext on a line plot
subrate.lplot <- function(AFR, EUR, EAS, SAS, mut){
  i <- which(AFR$X3mer == mut)
  
  AFR.dat <- cbind(AFR$Rate[i],AFR$Context[i], rep("AFR", 16))
  colnames(AFR.dat)<- c("Rate", "Context", "Pop")
  EUR.dat <- cbind(EUR$Rate[i],EUR$Context[i], rep("EUR", 16))
  colnames(EUR.dat)<- c("Rate", "Context", "Pop")
  EAS.dat <- cbind(EAS$Rate[i],EAS$Context[i], rep("EAS", 16))
  colnames(EAS.dat)<- c("Rate", "Context", "Pop")
  SAS.dat <- cbind(SAS$Rate[i],SAS$Context[i], rep("SAS", 16))
  colnames(SAS.dat)<- c("Rate", "Context", "Pop")
  
  data <- data.frame(rbind(AFR.dat, EUR.dat, EAS.dat, SAS.dat))
  
  myplot <- ggplot(data, aes(Pop, as.numeric(as.character(Rate)), group = Context, color = Context)) +
    geom_point(size = 4) +
    geom_line(size = 1.2) +
    labs(y = paste("Estimated mutation rate of ", mut,"\n")) + #y axis label
    theme(axis.text.x = element_text(size = rel(1.4)), axis.title.x = element_blank(), #adjust text sizes
          axis.title.y = element_text(size = rel(1.5)), axis.text.y = element_text(size = rel(1.4)))
  
  return(myplot)
}
```

# Mutation rate differences at the 3mer level

```{r upload significant, echo = F, message = F}
setwd("C:/Users/VoightLab/Dropbox/SNP_rates/Analyses/chi_squared_tests/results/ordered_p")
sig.3mers <- read_delim("o_significant_3mers.txt", 
                        delim ="\t", escape_double = F, trim_ws = T)
sig.5mers <- read_delim("o_significant_new_5mers.txt", 
                        delim ="\t", escape_double = F, trim_ws = T)
sig.7mers <- read_delim("o_significant_new_7mers.txt", 
                        delim ="\t", escape_double = F, trim_ws = T)
```

Recall the following list of the top 10 most highly significant 3mers:

```{r print sig.3mers, echo = F}
kable(head(sig.3mers, n = 14), row.names = F, digits = 600,
      caption = "14 most highly significant 3mers")
```

Here, I've chosen to show the top 14 results, which are all significant at p < 10^-40^.

\pagebreak

##Signal 1: European C->T Elevation

Among the top polymorphisms, TCC->T, ACC->T, TCT->T, and CCC->T have already been noted as part of the signal of European C->T enrichment. In addition, #11 and #12, TCA->T and ACT->T share the same profile as the other Europe-enriched C->T mutations and cluster together with them in heatmap experiments. However, they have not been noted in previous studies.  These two mutations are shown below, along with TCC->T for reference:

```{r CI signal1, echo = F, fig.width = 7, fig.height = 2.3}
a <- CI.plot(AFR_3mer_counts, EUR_3mer_counts, EAS_3mer_counts, SAS_3mer_counts, "TCC->T")
b <- CI.plot(AFR_3mer_counts, EUR_3mer_counts, EAS_3mer_counts, SAS_3mer_counts, "TCA->T")
c <- CI.plot(AFR_3mer_counts, EUR_3mer_counts, EAS_3mer_counts, SAS_3mer_counts, "ACT->T")
grid.arrange(a,b,c, ncol = 3)
```

```{r, chrom signal1, echo = F, fig.width = 7, fig.height = 2.3}
d <- chrom.box(AFR_3mer_counts, EUR_3mer_counts, EAS_3mer_counts, SAS_3mer_counts,
               gw_3mer_counts,"TCC->T")
e <- chrom.box(AFR_3mer_counts, EUR_3mer_counts, EAS_3mer_counts, SAS_3mer_counts,
               gw_3mer_counts,"TCA->T")
f <- chrom.box(AFR_3mer_counts, EUR_3mer_counts, EAS_3mer_counts, SAS_3mer_counts,
               gw_3mer_counts,"ACT->T")
grid.arrange(d,e,f, ncol = 3)
```
Note that although these patterns appear similar in EUR, EAS, and SAS, there are noteable differences between the relative rates in Africa. This seems to be common among the C->T signal polymorphisms. Consider for example CCC->T:

```{r CI CCCT, echo = F, fig.height=2.5, fig.width=2.3, fig.align="center"}
CI.plot(AFR_3mer_counts, EUR_3mer_counts, EAS_3mer_counts, SAS_3mer_counts,"CCC->T")
```

## Signal 2: Enrichment of certain polymorphisms in Europe

The fourth and fifth most significant results, however, GAT->T and ACC->A have not previously been noted, and are highly significant at p < 10^-85^. In addition, these mutation types are clustered with GAC->T, the 16^th^ most highly significantly heterogeneous polymorphism type.  The mutation rates for the three of these polymorphisms are shown below:

```{r CI signal2, echo = F, fig.width = 7, fig.height = 3}
a <- CI.plot(AFR_3mer_counts, EUR_3mer_counts, EAS_3mer_counts, SAS_3mer_counts, "GAT->T")
b <- CI.plot(AFR_3mer_counts, EUR_3mer_counts, EAS_3mer_counts, SAS_3mer_counts, "ACC->A")
c <- CI.plot(AFR_3mer_counts, EUR_3mer_counts, EAS_3mer_counts, SAS_3mer_counts, "GAC->T")
grid.arrange(a,b,c, ncol = 3)
```


```{r, chrom signal2, echo = F, fig.width = 7, fig.height = 3}
d <- chrom.box(AFR_3mer_counts, EUR_3mer_counts, EAS_3mer_counts, SAS_3mer_counts,
               gw_3mer_counts,"GAT->T")
e <- chrom.box(AFR_3mer_counts, EUR_3mer_counts, EAS_3mer_counts, SAS_3mer_counts,
               gw_3mer_counts,"ACC->A")
f <- chrom.box(AFR_3mer_counts, EUR_3mer_counts, EAS_3mer_counts, SAS_3mer_counts,
               gw_3mer_counts,"GAC->T")
grid.arrange(d,e,f, ncol = 3)
```
\pagebreak

## Signal 3: CpG transitions

All four CpG transitions appear among the top 11 most significant results.  Ian Mathieson and David Riech have previously noted that there is some amount of variation in CpG enrichment between populations, but that this variation is slight relative to the overall rate of CpG transitions. Other studies have noted that CpGs are the most "clocklike" polymorphism types among humans and other primates.  The CpG mutations are shown below:

```{r CI signal3, echo = F, fig.width = 5.5, fig.height = 6, fig.align="center"}
a <- CI.plot(AFR_3mer_counts, EUR_3mer_counts, EAS_3mer_counts, SAS_3mer_counts, "TCG->T")
b <- CI.plot(AFR_3mer_counts, EUR_3mer_counts, EAS_3mer_counts, SAS_3mer_counts, "ACG->T")
c <- CI.plot(AFR_3mer_counts, EUR_3mer_counts, EAS_3mer_counts, SAS_3mer_counts, "GCG->T")
d <- CI.plot(AFR_3mer_counts, EUR_3mer_counts, EAS_3mer_counts, SAS_3mer_counts, "CCG->T")
grid.arrange(a,b,c, d, ncol = 2)
```

It is worth nothing that the CpG mutations cluster together in heatmaps, even after normalization.

\pagebreak

##Signal 4: Heterogeneity within East Asia

When we construct a heatmap of all 3mers, we find two clusters which appear enriched in Japan and other groups in East Asia.  These clusters are comprised of the \*AC->C polymorphisms, as well as TAT->T. When the mutational types are clustered using only the data from East Asia, excluding other continental groups, we find that these two clusters merge, and the additional polymorphism CAC->C is added.  This group is in correspondence with results from Harris and Pritchard, who find that *AC->C, TAT->T, and CAC->C mutation types separate East Asians in a principal component analysis.

These mutation types have the global profiles shown below:

```{r upload 3mer subpops, echo = F, message = F}
source("C:/Users/VoightLab/Dropbox/SNP_rates/Code/data_wrangling/upload_subpops_3mer.R")
```

```{r, signal4 CI, echo =F}
CI.plot.subpop(subpops.3mer.ancestral, "AAC->C", pops.ancestral)
```

\pagebreak

## Additional Signals

Two remaining polymorphism types have not yet been mentioned. The first highly significant signal which has not been is ACA->T, which appears elevated in Africa.

```{r ACAT, echo = F, fig.align="center", fig.height=3, fig.width=5.5}
a <- CI.plot(AFR_3mer_counts, EUR_3mer_counts, EAS_3mer_counts, SAS_3mer_counts, "ACA->T")
b <- chrom.box(AFR_3mer_counts, EUR_3mer_counts, EAS_3mer_counts, SAS_3mer_counts,
               gw_3mer_counts,"ACA->T")
grid.arrange(a,b, ncol = 2)
```

The second is GCC->T,which is elevated in East Asia in addition to Europe and South Asia.


```{r GCCT, echo = F, fig.align="center", fig.height=3, fig.width=5.5}
a <- CI.plot(AFR_3mer_counts, EUR_3mer_counts, EAS_3mer_counts, SAS_3mer_counts, "GCC->T")
b <- chrom.box(AFR_3mer_counts, EUR_3mer_counts, EAS_3mer_counts, SAS_3mer_counts,
               gw_3mer_counts,"GCC->T")
grid.arrange(a,b, ncol = 2)
```

Finally, an additional pattern of interest is the shared profiles of certain CpG transversions, which appear to be enriched in Africa, and which cluster together in  heatmaps.  However, none of these polymorphism types are significant based on homogeneity tests (predictably, since CpG transversions are rare), and Harris and Pritchard have noted that the proportions of CpG transversions in 1,000 genomes and the Simmons Diversity Genome Project dataset tend not to agree, suggesting that this pattern may be driven by some sequencing artifact.

# Heterogeneity of 3mer signals within higher order sequence context models

Now that we have identified several groups of 3mer polymorphisms which appear to vary across populations, we would like to know whether local sequence context (up to 3 bases from the substitution) plays a role in driving the variation we observe.  To do this, we can bin the polymorphisms from any given 3mer (say, TCC->T) into 256 different 7mer expansions (e.g. ACTCCCT->A), and observe how their rates differ between populations.  

```{r upload 7mer subpops, echo = F, message = F}
source("C:/Users/VoightLab/Dropbox/SNP_rates/Code/data_wrangling/upload_subpops_7mer.R")
```


We'll begin with a simple null example.  The CpG transition CCG->T is relatively the same within East Asia. In this case, we expect to see some variation in rate between CCG->T expanisions due to 7mer effects that are uniform across all populations.  However, we expect the rates of any given 7mer to be equal between East Asian subpopulations.  This is what we see below, in Chinese Dai versus Japanese from Tokyo:

```{r null heterogeneity plot, echo = F, fig.align="center", fig.width = 3.25, fig.height=3.25}
subrate.scplot(CDX_EAS_7mer_counts, JPT_EAS_7mer_counts, "CCG->T", c("CDX", "JPT"))
```
Here, each point represents a polymorphism, and the points are colored by the base 10 log of sample size (number of polymorphisms observed).  We see some noise here, mostly among the yellow-colored (more uncertain) polymorphisms.  However, most points lie along the y = x line.

## Signal 1: European C->T elevation

For each signal we've highlighted at the 3mer level, we'd like to know whether this is a true 3mer effect or whether this is driven by broader sequence context.  

Let's consider the top 3mer polymorphisms for the European C->T elevation, shown on the next page. Here, we can see that, for each polymorphism type, the distribution of all 7mers lies slightly above the y = x line.  This indicates that signal 1 is determined by local sequence context effects at 1 or fewer base pairs from the substitution locus.

\pagebreak

```{r signal1 heterogeneity, echo = F, fig.height=9}
a <- subrate.scplot(EAS_7mer_counts, EUR_7mer_counts, "TCC->T", pops = c("EAS", "EUR"))
b <- subrate.scplot(EAS_7mer_counts, EUR_7mer_counts, "ACC->T", pops = c("EAS", "EUR"))
c <- subrate.scplot(EAS_7mer_counts, EUR_7mer_counts, "TCT->T", pops = c("EAS", "EUR"))
d <- subrate.scplot(EAS_7mer_counts, EUR_7mer_counts, "CCC->T", pops = c("EAS", "EUR"))
e <- subrate.scplot(EAS_7mer_counts, EUR_7mer_counts, "ACT->T", pops = c("EAS", "EUR"))
f <- subrate.scplot(EAS_7mer_counts, EUR_7mer_counts, "TCA->T", pops = c("EAS", "EUR"))
grid.arrange(a,b,c,d,e,f, ncol = 2)
```
 
\pagebreak 
 
## Signal 2: Enrichment of certain polymorphisms in East Asia

We come to a similar conclusion for Signal 2:

```{r signal2 heterogeneity, echo = F, fig.height=6}
a <- subrate.scplot(AFR_7mer_counts, EAS_7mer_counts, "GAT->T", pops = c("AFR", "EAS"))
b <- subrate.scplot(AFR_7mer_counts, EAS_7mer_counts, "ACC->A", pops = c("AFR", "EAS"))
c <- subrate.scplot(AFR_7mer_counts, EAS_7mer_counts, "GAC->T", pops = c("AFR", "EAS"))
grid.arrange(a,b,c, ncol = 2)
```


\pagebreak 

## Signal 3: CpG polymorphisms

SAS vs AFR.  Unsurprisingly, this appears to be a 3mer-level signal.

```{r signal3 heterogeneity, echo = F, fig.height=6}
a <- subrate.scplot(AFR_7mer_counts, SAS_7mer_counts, "TCG->T", pops = c("AFR", "SAS"))
b <- subrate.scplot(AFR_7mer_counts, SAS_7mer_counts, "ACG->T", pops = c("AFR", "EAS"))
c <- subrate.scplot(AFR_7mer_counts, SAS_7mer_counts, "GCG->T", pops = c("AFR", "EAS"))
d <- subrate.scplot(AFR_7mer_counts, SAS_7mer_counts, "CCG->T", pops = c("AFR", "EAS"))
grid.arrange(a,b,c, d, ncol = 2)
```


\pagebreak

## Signal 4: Heterogeneity within East Asia

Now we shift to discussing the fourth signal: *AC->C and TAT->T, which appear to be elevated in East Asia, most notably in certain individuals from Japan and China.  In order to understand how this mutation type varies within East Asia, I will plot the rates of these polymorphisms in Japan versus Chinese Dai in Xishuangbana. These can be seen on the following page. In contrast to the previous plots, most points lie along the line y = x, with a few outliers.  This indicates that some cues among the 7mer sequence context may be important.

Intrigued by these findings, we set out to begin to identify putative 7mer types responsible for this signature. To this end, we considered each of the 1280 possible 7mer expansions *AC->C and TAT->T 3-mer substitutions, testing for heterogeneity between Japanese from Tokyo (JPT, higher signature 4 polymorphism proportion) and Chinese Dai from Xishuangbana (CDX, lower signature 4 polymorphism proportion).

```{r JPTCDX 7mers test, echo = F}
p.CDXJPT <- pairwise.chi(CDX_EAS_7mer_counts, JPT_EAS_7mer_counts)
p.CDXJPT <- subset(p.CDXJPT, is.element(CDX_EAS_7mer_counts$X3mer, c("AAC->C", "CAC->C", "TAT->T", "GAC->C", "TAC->C")))
p.CDXJPT$fdr <- p.adjust(p.CDXJPT$p, method = "fdr")
kable(subset(p.CDXJPT, p.CDXJPT$fdr < 0.05), row.names = F)
```

Among these polymorphisms, the motif ACAG->C appears to be very common.

```{r def preprocessor, echo = F}
X.preprocessing <- function(counts, gw, muts = c()){
  
  # if muts not specified, assume whole dataframe is used
  if (length(muts) == 0){
    muts <- counts$Context
  }
  counts <- subset(counts, is.element(counts$Context, muts))
  autosomes <-counts$Count-counts$chrX
  
  # construct data frame
  result <- data.frame(matrix(ncol=3,nrow=length(counts$Context)))
  colnames(result) <- c("Context", "Autosomes", "X")
  result$Context <- counts$Context
  result$Autosomes <- autosomes
  result$X <- counts$chrX
  
  # add gw_sites to data
  result$Autosomal_sites <- rep(0, length(result$Context))
  result$X_sites <- rep(0, length(result$Context))
  
  for (i in 1:length(result$Context)){
    mut <- as.character(result$Context[i])
    context <- substr(mut, 1, nchar(mut)-3)
    j <- which(gw$Context == context)
    result$Autosomal_sites[i] <- gw$GW_total[j]- gw$X[j]
    result$X_sites[i] <- gw$X[j]
  }

  return(result)
}
```

Testing for enrichment on the X chromosome, we find that 4 of 9 of these polymorphisms is significantly enriched on X (see below).  The probability of observing this number of significant results by chance alone is:

```{r}
binom.test(4,9,0.05)$p.value
```

```{r, setup for X test, echo = F}
JPT7mers <- subset(p.CDXJPT$Context, p.CDXJPT$fdr < 0.05)
data <- X.preprocessing(EAS_7mer_counts, gw_7mer_counts, c())
```

```{r x test, echo = F}
data$alpha <- rep(1, length(data$Context))
data$p.0 <- rep(0, length(data$Context))
data$p.MLE <- rep(0, length(data$Context))
data$p <- rep(1, length(data$Context))

muts.X <- sum(data$X)
sites.X <- sum(data$X_sites)
muts.A <- sum(data$Autosomes)
sites.A <- sum(data$Autosomal_sites)

data <- subset(data, is.element(data$Context, JPT7mers))

#remove the data from these sites from background before estimating correction
muts.X <- muts.X - sum(data$X)
sites.X <- sites.X - sum(data$X_sites)
muts.A <- muts.A - sum(data$Autosomes)
sites.A <- sites.A - sum(data$Autosomes)

alpha <- (muts.X/sites.X)/(muts.A/sites.A)

for (i in 1:length(data$Context)){
  # estimate parameters and run test
  data$alpha[i] <- alpha
  p.0 <- alpha*(data$Autosomes[i]/data$Autosomal_sites[i])
  data$p.0[i] <- p.0
  data$p.MLE[i] <- data$X[i]/data$X_sites[i]
  
  data$p[i] <- binom.test(data$X[i], n = data$X_sites[i], p = p.0, alternative = "greater")$p.value
  
  if (data$Autosomes[i]< 15){
    data$p[i] <- 1
  }
  
}
kable(data, digits = 6, row.names = F)
```

```{r plot signal4 heterogeneity, echo = F, fig.height=9}
a <- subrate.scplot(CDX_EAS_7mer_counts, JPT_EAS_7mer_counts, "AAC->C", pops = c("Chinese Dai", "Japanese"), label = JPT7mers)
b <- subrate.scplot(CDX_EAS_7mer_counts, JPT_EAS_7mer_counts, "CAC->C", pops = c("Chinese Dai", "Japanese"), label = JPT7mers)
c <- subrate.scplot(CDX_EAS_7mer_counts, JPT_EAS_7mer_counts, "GAC->C", pops = c("Chinese Dai", "Japanese"), label = JPT7mers)
d <- subrate.scplot(CDX_EAS_7mer_counts, JPT_EAS_7mer_counts, "TAC->C", pops = c("Chinese Dai", "Japanese"), label = JPT7mers)
e <- subrate.scplot(CDX_EAS_7mer_counts, JPT_EAS_7mer_counts, "TAT->T", pops = c("Chinese Dai", "Japanese"), label = JPT7mers)
grid.arrange(a,b,c,d,e, ncol = 2)
```


## Additional 3mer signals

Here is ACA->T

```{r ACAT heterogeneity, echo = F, fig.height=3}
subrate.scplot(EAS_7mer_counts, AFR_7mer_counts, "ACA->T", pops = c("EAS", "AFR"))
```

This appears also to be a 3mer-level signal, although three mutation types at the bottom right appear to be outliers.

*****

The last signal is GCC->T:
```{r GCCT heterogeneity, echo = F, fig.height=3, fig.align="center"}
subrate.scplot(AFR_7mer_counts, EUR_7mer_counts, "GCC->T", pops = c("AFR", "EUR"))
```

This one appears also to be a 3mer signal, but the whole story may be more complicated, since we are seeing some abnormal results at the 5mer level (see notedbook).

# Novel mutation rate differences at the 7mer level

After removing known 3mer signals (TCC->T, ACC->T, TCT->T, ATC->C, ACC->C, and GAT->T), we have the following top significant results for 7mers:

```{r significant 7mers, echo = F}
kable(head(sig.7mers[,-c(2,4)], n = 10), row.names = F, digits = 600,
      caption = "10 most highly significant 7mers, after removing top 3mer signals")
```

## Additional polymorphisms heterogeneous in East Asia

Three of these, CAAACCC->C, TTTATTT->T, and AGTACAG->C are among the 7mers which we observe to be enriched in Japan.  Interestingly, another two also appear enriched in Japan: AAACAAA->A, and CTGCATA->G.  Their gloabel profiles are shown below.

```{r AAACAAA, echo = F, fig.height = 3}
CI.plot.subpop(subpops.7mer.ancestral, "AAACAAA->A", pops.ancestral, colors.ancestral)
```

\pagebreak
 
## TAA->T

The remaining polymorphisms are all within A->T rich contexts.  Here we examine the first, a pair of polymorphisms with the 3mer subcontext TAA->T.

```{r TAAT, echo = F, fig.height=2.5}
#TAA->T 3mer subcontexts
a <- CI.plot(AFR_3mer_counts, EUR_3mer_counts, EAS_3mer_counts, 
             SAS_3mer_counts, "TAA->T")
b <- CI.plot(AFR_7mer_counts, EUR_7mer_counts, EAS_7mer_counts, 
             SAS_7mer_counts, "TTTAAAA->T")
c <- CI.plot(AFR_7mer_counts, EUR_7mer_counts, EAS_7mer_counts, 
             SAS_7mer_counts, "ATTAAAA->T")
grid.arrange(a,b,c, ncol = 3)
```

For the most part, the profiles of TTTAAAA->T and ATTAAAA->T match that of the broader 3mer subcontext.  However, upon closer examination we find that TTTAAAA->T may be outlier contexts.  For most other TAA->T exansions, in fact, the rates in Africa are in agreement with the rates in Europe.  Only these two highly variable contexts appear to be driving the signal on the 3mer level.

```{r heterogeneity TAAT, echo = F, fig.height=3, fig.align="center"}
subrate.scplot(EUR_7mer_counts, AFR_7mer_counts, "TAA->T", pops = c("EUR", "AFR"), label = c("TTTAAAA->T", "ATTAAAA->T"))
```

\pagebreak

## TAA->G

Surprisingly, TAA->G has an altogether different pattern than TAA->T.  Again, we see that the 3mer subcontext is more or less in agreement with the profile of this 7mer expansion.

```{r, echo = F, fig.align='center', fig.height=2.5, fig.width=4.5}
#TAA->G looks different from TAA->T
a <- CI.plot(AFR_7mer_counts, EUR_7mer_counts, EAS_7mer_counts, 
             SAS_7mer_counts, "ACTAAAA->G")
b <- CI.plot(AFR_3mer_counts, EUR_3mer_counts, EAS_3mer_counts, 
             SAS_3mer_counts, "TAA->G")
grid.arrange(a,b,ncol = 2)
```

```{r TAAG heterogeneity, echo = F, fig.align="center", fig.height=3}
subrate.scplot(AFR_7mer_counts, EAS_7mer_counts, "TAA->G", pops = c("AFR", "EAS"))
```

Based on the scatterplot above, it is possible that the profile of TAA->G is actually shaped by a small handful of 7mer outliers.

## Remaining TAT signals

```{r, echo = F, fig.align='center', fig.height=2.5, fig.width=4.5}
#TATATAT->G does not resemble 3mer subcontext
a <- CI.plot(AFR_7mer_counts, EUR_7mer_counts, EAS_7mer_counts, 
             SAS_7mer_counts, "TATATAT->G")
b <- CI.plot(AFR_3mer_counts, EUR_3mer_counts, EAS_3mer_counts, 
             SAS_3mer_counts, "TAT->G")
grid.arrange(a,b,ncol = 2)
```

```{r ATAC heterogeneity, echo = F, fig.align="center", fig.height=3}
subrate.scplot(EAS_7mer_counts, AFR_7mer_counts, "TAT->G", pops = c("EAS", "AFR"))
```

To be honest, I'm not really sure what the correct interpretation of this is.

```{r, echo = F, fig.align='center', fig.height=2.5, fig.width=4.5}
#TATATTT->T dpes not resemble its 3mer subcontext
a <- CI.plot(AFR_7mer_counts, EUR_7mer_counts, EAS_7mer_counts, 
             SAS_7mer_counts, "TATATTT->T")
b <- CI.plot(AFR_3mer_counts, EUR_3mer_counts, EAS_3mer_counts, 
             SAS_3mer_counts, "TAT->T")
grid.arrange(a,b,ncol = 2)
```

```{r TATT heterogeneity, echo = F, fig.align="center", fig.height=3}
subrate.scplot(EAS_7mer_counts, AFR_7mer_counts, "TAT->T", pops = c("EAS", "AFR"))
```
